infrastructure/src/main/java/org/corfudb/infrastructure/LogUnitServer.java

```diff
@@ -253,8 +242,8 @@ private void fillHole(CorfuPayloadMsg<TrimRequest> msg, ChannelHandlerContext ct
 
     @ServerHandler(type = CorfuMsgType.TRIM)
     private void trim(CorfuPayloadMsg<TrimRequest> msg, ChannelHandlerContext ctx, IServerRouter r) {
-        trimMap.compute(msg.getPayload().getStream(), (key, prev) ->
-                prev == null ? msg.getPayload().getPrefix() : Math.max(prev, msg.getPayload().getPrefix()));
+        batchWriter.write(new LogAddress(msg.getPayload().getPrefix(), msg.getPayload().getStream()), null);
+        //TODO(Maithem): should we return an error if the write fails
```

Comment:
I think this comment can be removed. if it fails, it will throw an uncaught exception.